File: pinyougou-web-user/src/main/java/cn/itcast/core/controller/UserController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.address.Address;
import cn.itcast.core.pojo.user.User;
import cn.itcast.core.service.UserService;
import com.alibaba.dubbo.config.annotation.Reference;
import entity.Result;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import util.PhoneFormatCheckUtils;

import java.util.List;

/**
 * Created by wang on 2019/4/26.
 */
@RestController
@RequestMapping("/user")
public class UserController {
    @Reference
    private UserService userService;
    @RequestMapping("/sendCode")
    public Result sendConde(String phone){
        try {
            if (PhoneFormatCheckUtils.isPhoneLegal(phone)){
                userService.sendCode(phone);
                return new Result(true,"短信发送成功");
            }
            return new Result(false,"手机号格式不正确");
        }catch (Exception e){
            e.printStackTrace();
            return new Result(false,"短信发送失败");
        }
    }

    @RequestMapping("/add")
    public Result add(String smscode,@RequestBody User user){
        try {
            userService.add(smscode,user);
            return new Result(true,"注册成功");

        }catch (RuntimeException re){
            return new Result(false, re.getMessage());
        } catch (Exception e){
            e.printStackTrace();
            return new Result(false,"注册失败");
        }
    }
    @RequestMapping("/selectAddress")
    public List<Address> selectAddress(){
        String name = SecurityContextHolder.getContext().getAuthentication().getName();
        List<Address> addresses = userService.selectAddress(name);
        return addresses;
    }
    @RequestMapping("/addAddress")
    public Result addAddress(@RequestBody Address address){
        try {
            userService.addAddress(address);
            return new Result(true,"成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"失败");
        }
    }
    @RequestMapping("/updateAddress")
    public Result updateAddress(@RequestBody Address address){
        try {
            userService.updateAddress(address);
            return new Result(true,"成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"失败");
        }
    }
    @RequestMapping("/deleteAddress")
    public Result deleteAddress(Long id){
        try {
            userService.deleteAddress(id);
            return new Result(true,"成功");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false,"失败");
        }
    }
    @RequestMapping("/findOneAddress")
    public Address findOneAddress(Long id){
      return   userService.findOneAddress(id);
    }

}

File: pinyougou-web-user/src/main/webapp/js/controller/userController.js
 //控制层 
app.controller('userController' ,function($scope,$controller   ,userService){	
	$scope.entity = {};
	//注册用户
	$scope.reg=function(){
		
		//比较两次输入的密码是否一致
		if($scope.password!=$scope.entity.password){
			alert("两次输入密码不一致，请重新输入");
			$scope.entity.password="";
			$scope.password="";
			return ;			
		}
		//新增
		userService.add($scope.entity,$scope.smscode).success(
			function(response){
				alert(response.message);
			}		
		);
	}
    
	//发送验证码
	$scope.sendCode=function(){
		if($scope.entity.phone==null || $scope.entity.phone==""){
			alert("请填写手机号码");
			return ;
		}
		
		userService.sendCode($scope.entity.phone  ).success(
			function(response){
				alert(response.message);
			}
		);		
	}
	$scope.selectAddress=function () {
		userService.selectAddress().success(
            function(response){
                $scope.list=response;
            }
        );
    }
    //删除 地址
    $scope.deleteAddress=function(id){
        //获取选中的复选框
        userService.deleteAddress(id).success(
            function(response){
                if(response.flag){
                    alert(response.message);
                    $scope.selectAddress();//刷新列表
                }else {
                    alert(response.message);
				}
            }
        );
    }

    $scope.saveAddress=function(){
        var serviceObject;//服务层对象
        if($scope.entity.id!=null){//如果有ID
            serviceObject=userService.updateAddress( $scope.entity ); //修改
        }else{
            serviceObject=userService.addAddress( $scope.entity  );//增加
        }
        serviceObject.success(
            function(response){
                if(response.flag){
                    //重新查询
                    $scope.selectAddress();//重新加载
                }else{
                    alert(response.message);
                }
            }
        );
    }

    // 查询一个:
    $scope.findOneAddress = function(id){
        userService.findOneAddress(id).success(function(response){

            $scope.entity = response;
        });
    }

	
});	
